packages/core-data/src/utils/test/get-nested-value.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Internal dependencies
+ */
+import getNestedValue from '../get-nested-value';
+
+describe( 'getNestedValue', () => {
+	it( 'should return the same object unmodified if path is an empty array', () => {
+		const input = { x: 'y' };
+		const result = getNestedValue( input, [] );
+		expect( result ).toEqual( input );
+	} );
+
+	it( 'should the nested value', () => {
+		const input = { x: { y: { z: 123 } } };
+		const result = getNestedValue( input, [ 'x', 'y', 'z' ] );
+
+		expect( result ).toEqual( 123 );
+	} );
+
+	it( 'should return the nested value if the path is a string', () => {
+		const input = { x: { y: { z: 123 } } };
+		const result = getNestedValue( input, 'x.y.z' );
+
+		expect( result ).toEqual( 123 );
+	} );
+
+	it( 'should return the shallow value', () => {
+		const input = { x: { y: { z: 123 } } };
+		const result = getNestedValue( input, 'x' );
+
+		expect( result ).toEqual( { y: { z: 123 } } );
+	} );
+
+	it( 'should return the default value if the nested value is undefined', () => {
+		const input = { x: { y: { z: undefined } } };
+		const result = getNestedValue( input, [ 'x', 'y', 'z' ], 456 );
+
+		expect( result ).toEqual( 456 );
+	} );
+
+	it( 'should return the nested value if it is different to undefined', () => {
+		const input = { x: { y: { z: null } } };
```

Comment:
This should use a different value than `null` if we end up returning the default value for "nullish" values.

Reply:
Nope we will return `null` is thee value is `null`. I corrected the function jsdoc comment: https://github.com/WordPress/gutenberg/pull/54161/commits/de832bf14a6789cdf0bec15c5117799b14622b7c